pkg/skaffold/survey/survey.go

```diff
@@ -74,20 +90,32 @@ func isSurveyPromptDisabled(configfile string) (*sConfig.GlobalConfig, bool) {
 		*cfg.Global.Survey.DisablePrompt
 }
 
-func recentlyPromptedOrTaken(cfg *sConfig.GlobalConfig) bool {
+func (s *Runner) recentlyPromptedOrTaken(cfg *sConfig.GlobalConfig) bool {
 	if cfg == nil || cfg.Global == nil || cfg.Global.Survey == nil {
-		return false
+		return s.taken(cfg.Global.Survey)
 	}
-	return timeutil.LessThan(cfg.Global.Survey.LastTaken, 90*24*time.Hour) ||
-		timeutil.LessThan(cfg.Global.Survey.LastPrompted, 10*24*time.Hour)
+	return recentlyPrompted(cfg.Global.Survey) || s.taken(cfg.Global.Survey)
 }
 
-func (s *Runner) DisplaySurveyPrompt(out io.Writer) error {
+func recentlyPrompted(gc *sConfig.SurveyConfig) bool {
+	return timeutil.LessThan(gc.LastPrompted, 10*24*time.Hour)
+}
+
+func (s *Runner) taken(gc *sConfig.SurveyConfig) bool {
```

Comment:
this method initializes `s.surveyID` as a side-effect. Can we initialize it in the `New` method instead?

Reply:
I can make this better.